widgetboxmodel/src/frontendwidget.cpp

```diff
@@ -540,20 +540,22 @@ void WidgetBoxModel::FrontendWidget::resizeEvent(QResizeEvent *event) {
     d->settingsButton_->move(d->ui.frame->geometry().topRight() - QPoint(d->settingsButton_->width()-1,0));
 
 #ifdef __unix__
-    // Keep the input shape consistent
-    int shape_event_base, shape_error_base;
-    if (XShapeQueryExtension(QX11Info::display(), &shape_event_base, &shape_error_base)) {
-
-        Region region = XCreateRegion();
-        XRectangle rectangle;
-        int scalefactor = devicePixelRatio();  // TODO Qt>5.6 devicePixelRatioF
-        rectangle.x      = static_cast<int16_t>(d->ui.frame->geometry().x()*scalefactor);
-        rectangle.y      = static_cast<int16_t>(d->ui.frame->geometry().y()*scalefactor);
-        rectangle.width  = static_cast<uint16_t>(d->ui.frame->geometry().width()*scalefactor);
-        rectangle.height = static_cast<uint16_t>(d->ui.frame->geometry().height()*scalefactor);
-        XUnionRectWithRegion(&rectangle, region, region);
-        XShapeCombineRegion(QX11Info::display(), winId(), ShapeInput, 0, 0, region, ShapeSet);
-        XDestroyRegion(region);
+    if (QGuiApplication::platformName() != "wayland") {
```

Comment:
It seems like the "better" way to do this is not to "disable X11-hacks on Wayland" but to "enable X11-hacks on X11". I would vote to close this PR in favor of #21 which has a (in my opinion) better guard clause.